src/ucp/rma/flush.c

```diff
@@ -444,16 +444,17 @@ static unsigned ucp_worker_flush_progress(void *arg)
     ucp_ep_h ep;
 
     status = ucp_worker_flush_check(worker);
-    if ((status == UCS_OK) || (&next_ep->ep_list == &worker->all_eps)) {
+    if ((status == UCS_OK) && (&next_ep->ep_list == &worker->all_eps)) {
```

Comment:
isn't this disabling endpoint-by-endpoint flush? if we always require iface flush to be completed?
maybe we can check just flush_ops_count?

Reply:
> isn't this disabling endpoint-by-endpoint flush?

sorry, do you mean that it could disable case when we all `uct_ep_flush()`s are completed with `UCS_OK`, but `uct_iface_flush()` returns `UCS_ERR_NO_RESOURCE`?